scripts/shared/lib/clusters_kind

```diff
@@ -281,7 +282,7 @@ function download_ovnk() {
 
 function provider_prepare() {
     [[ -z "${K8S_VERSION}" ]] && K8S_VERSION="${DEFAULT_K8S_VERSION}"
-    kind="${kind_binaries[$K8S_VERSION]:-kind-0.12}"
+    kind="${kind_binaries[$K8S_VERSION]:-kind}"
```

Comment:
You should probably manually test `make e2e K8S_VERSION=...` on operator with different versions just to make sure it's fine with this change.